IanGRap/Squish
# Issue 192

Contributors: Ian

Goal: Fix issue where pushing blocks into walls as they compact causes the walls to wig out in real weird ways.

This bug was a real quick fix but super hard to recreate. Recreating the bug is an ordeal so be ready. 

Important Files:
- Wall.cs: Made it so that the walls break any blocks they touch when they are compacting.
- WallTester.cs: Inherits from Wall. Basically pushes blocks towards itself whenever it compacts. This is meant to make a block push into a wall the same second the wall compacts. This seems to be the only instance in which the bug happens and is really hard to recreate without a script. 

How to test:
- Comment out line 101 on Wall.cs so you can see the error.
- Load the Wall Test.unity scene. Press play (I recommend you disable the block spawner once the scene starts so you can just watch the walls). After the walls compact a few times they should be acting very strange/broken.
- Uncomment line 101 on Wall.cs. Run the scene again and it should be fine. 

Notes:
Should we delete the test files after this pr is approved or should we keep them in case this bug returns later?
